org.lflang.tests/src/org/lflang/tests/Configurators.java

```diff
@@ -46,24 +46,35 @@
     }
 
     /**
-     * Configure the given test by setting its `threads` target property to 0.
+     * Configure the given test to use single-threaded execution.
+     *
+     * For targets that provide a threaded and an unthreaded runtime,
+     * this configures using the unthreaded runtime. For targets that
+     * do not distinguish threaded and unthreaded runtime, the number
+     * of workers is set to 1.
      *
      * @param test The test to configure.
      * @return True if successful, false otherwise.
      */
     public static boolean useSingleThread(LFTest test) {
-        test.context.getArgs().setProperty("threads", "0");
+        test.context.getArgs().setProperty("threading", "false");
+        test.context.getArgs().setProperty("workers", "1");
         return true;
     }
 
     /**
-     * Configure the given test by setting its `threads` target property to 4.
+     * Configure the given test to use multi-threaded with 4 workers.
+     *
+     * For targets that provide a threaded and an unthreaded runtime,
+     * this configures using the threaded runtime. For all targets,
+     * the number of workers is set to 4.
      *
      * @param test The test to configure
      * @return True if successful, false otherwise.
      */
     public static boolean useFourThreads(LFTest test) {
-        test.context.getArgs().setProperty("threads", "4");
+        test.context.getArgs().setProperty("threading", "true");
```

Comment:
This preserves meaning, but can't we remove this method altogether?

Reply:
Never mind this. I thought we were running all the tests multi-threaded by default and had a special case for single-threaded, but the way it's setup currently we force all tests to be single-threaded and have multi-threaded as the special case...